enterprise/ha/src/main/java/org/neo4j/kernel/ha/UpdatePullingTransactionObligationFulfiller.java

```diff
@@ -70,8 +75,13 @@ public boolean evaluate( int currentTicket, int targetTicket )
                  * right after leaving this method we might read records off of disk, and they had better
                  * be up to date, otherwise we read stale data.
                  */
-                return transactionIdStore != null &&
-                       transactionIdStore.getLastClosedTransactionId() >= toTxId;
+                boolean condition = transactionIdStore != null &&
+                            transactionIdStore.getLastClosedTransactionId() >= toTxId;
+                if (condition)
+                {
+                    logger.debug( "Start obligation fulfill for txId=" + toTxId + " done, actual is txId=" + (transactionIdStore == null ? null : transactionIdStore.getLastClosedTransactionId() ) );
```

Comment:
transactionIdStore will never be null here, right? The condition will be false if so, therefore the if will not be entered